### PR TITLE
Add a CONTRIBUTING documentation to the pipeline-library project

### PR DESCRIPTION
**Pull request by aneveux**
Hello jenkins-infra team :wave: 

Here is a contribution proposal for the issue https://issues.jenkins.io/browse/INFRA-2881.

I created the CONTRIBUTING documentation based on information I found in the repository and some other contributing documentation from Jenkins projects.

Please let me know if you have any remarks or require any modification, I'll be happy to consider your guidance :)

Thanks a lot for your review, and have a great day!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Contributing to Pipeline Global Library for ci.jenkins.io
+
+👏 First off, thanks a lot for taking the time to contribute to this global library of pipeline components for [ci.jenkins.io](https://ci.jenkins.io/)!
```
Comment (olblak) [inline CONTRIBUTING.md:3]: This library is used by all of our Jenkins instances
* infra.ci.jenkins.io
* release.ci.jenkins.io
* trusted.ci.jenkins.io
* ci.jenkins.io
:)

**Review by olblak — APPROVED** [review]
That's really awesome and already a nice step forward

File: CONTRIBUTING.md (hunk 1 above)
Comment (dduportal) [reply CONTRIBUTING.md:3]: WDYT of the following?

```suggestion
👏 First off, thanks a lot for taking the time to contribute to this global library of pipeline components for [ci.jenkins.io](https://ci.jenkins.io/) (and many more Jenkins controllers of the [Jenkins Infrastructure Platform](https://www.jenkins.io/projects/infrastructure/)!
```

**Review by dduportal — DISMISSED** [review]
Thanks a lot! That is really cool!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+### Technical Requirements
+
+In order to work with that repository you will need:
+
+- OpenJDK 8: https://openjdk.java.net/projects/jdk8/
+- Maven 3.6+: https://maven.apache.org/download.cgi
+- Groovy: https://groovy-lang.org/install.html
+
+An IDE or text editor of your choice, like [Vim](https://www.vim.org/), [Emacs](https://www.vim.org/) or [VS Code](https://www.vim.org/) for example.
```
Comment (jglick) [inline CONTRIBUTING.md:105]: Copy-paste error, or start of flame war? :grin:

Comment (MarkEWaite) [inline CONTRIBUTING.md:105]: Love the comment from Jesse.  Even better when I entered "emacs" into a google search, their advisor suggested, "Did you mean: **vi**".  When I enter "vi" into a google search, their advisor suggests, "Did you mean **emacs**".  At least the Google search engine is trolling both camps equally.
```suggestion
An IDE or text editor of your choice, like [Vim](https://www.vim.org/), [Emacs](https://www.gnu.org/software/emacs/) or [VS Code](https://code.visualstudio.com/) for example.
```

**Review by olblak — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (aneveux) [reply CONTRIBUTING.md]: It was my gentle troll/easter egg for someone to find :p 

Now that you did I'll fix it to the proper links ;)

Comment (aneveux) [reply CONTRIBUTING.md]: Always the kind of little joke that makes me laugh :D

**Review by dduportal — APPROVED** [review]

**Pull request merged**